src/index.ts

```diff
@@ -1,18 +1,28 @@
 import { onSnapshot, applySnapshot } from 'mobx-state-tree'
 
-import * as LocalStorage from './localStorageAdaptor.js'
+import AsyncLocalStorage from './asyncLocalStorage'
 
-export const persist = (name, store, options = {}) => {
+export interface IArgs {
+  (name: string, store: object, options?: IOptions): void
+}
+export interface IOptions {
+  storage?: any,
+  jsonify?: boolean,
+  readonly whitelist?: Array<string>,
+  readonly blacklist?: Array<string>
+}
+
+export const persist: IArgs = (name, store, options = {}) => {
   let {storage, jsonify, whitelist, blacklist} = options
 
   if (typeof window.localStorage !== 'undefined' && (!storage || storage === window.localStorage)) {
-    storage = LocalStorage
+    storage = AsyncLocalStorage
   }
   if (!jsonify) { jsonify = true } // default to true like mobx-persist
   const whitelistDict = arrToDict(whitelist)
   const blacklistDict = arrToDict(blacklist)
 
-  onSnapshot(store, (_snapshot) => {
+  onSnapshot(store, (_snapshot: any) => {
```

Comment:
I was thinking of making this of type `StrToAnyMap` to resolve the implicit `any`, but honestly not sure if the keys will always be of type `string`. They _should_ be afaik, though not sure if they always will. In MST the type [is a generic](https://github.com/mobxjs/mobx-state-tree/blob/e9b711630c93e3212c28d83b8d0a82eb9763ec55/packages/mobx-state-tree/src/core/mst-operations.ts#L104).

Reply:
Well this is an internal type not exposed in the typings, so should be fine to type it as `StrToAnyMap`